modules/i18n/input-method/default.nix

```diff
@@ -0,0 +1,67 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+let
+  cfg = config.i18n.inputMethod;
+
+  gtk2_cache = pkgs.runCommandLocal "gtk2-immodule.cache" {
+    buildInputs = [ pkgs.gtk2 cfg.package ];
+  } ''
+    mkdir -p $out/etc/gtk-2.0/
+    GTK_PATH=${cfg.package}/lib/gtk-2.0/ gtk-query-immodules-2.0 > $out/etc/gtk-2.0/immodules.cache
+  '';
+
+  gtk3_cache = pkgs.runCommandLocal "gtk3-immodule.cache" {
+    buildInputs = [ pkgs.gtk3 cfg.package ];
+  } ''
+    mkdir -p $out/etc/gtk-3.0/
+    GTK_PATH=${cfg.package}/lib/gtk-3.0/ gtk-query-immodules-3.0 > $out/etc/gtk-3.0/immodules.cache
+  '';
+
+in {
+  imports =
+    [ ./fcitx.nix ./fcitx5.nix ./hime.nix ./kime.nix ./nabi.nix ./uim.nix ];
+
+  options.i18n = {
+    inputMethod = {
+      enabled = mkOption {
+        type = types.nullOr
+          (types.enum [ "fcitx" "fcitx5" "nabi" "uim" "hime" "kime" ]);
+        default = null;
+        example = "fcitx";
+        description = ''
+          Select the enabled input method. Input methods is a software to input symbols that are not available on standard input devices.
+
+          Input methods are specially used to input Chinese, Japanese and Korean characters.
+
+          Currently the following input methods are available in Home Manager:
+
+          <itemizedlist>
```

Comment:
I think a `<variablelist>` would be more suitable. Something like

```xml
<variablelist>
<varlistentry>
  <term><literal>fcitx</literal></term>
  <listitem><para>
    A customizable lightweight input method, extra input
    engines can be added using
    <literal>i18n.inputMethod.fcitx.engines</literal>.
  </para></listitem>
</varlistentry>
…
</variablelist>
```

See [systemd.user.startServices](https://rycee.gitlab.io/home-manager/options.html#opt-systemd.user.startServices) for an existing example.

Reply:
Done.